### DISCUSSION TITLE
prevent basePath rewrite the root url by using "rewrites" in next.config.js

### DISCUSSION THREAD

**Discussion by mtin79** [vercel/next.js#17315] [Help]
I need the basePath option in production to rewrite all my Link, Route, generated urls in pages and components.
But the root index page should remain as it is (as i am hosting a running next.js application on a subPath via nginx reverse proxy).

**The following next.config.js throws the error "Invalid rewrite found" :**

```
module.exports = {
  basePath: process.env.NODE_ENV === "production" ? "/test" : "",
  assetPrefix: process.env.NODE_ENV === "production" ? "/test/" : "",
  async rewrites() {
    if (process.env.NODE_ENV === "production") {
      return [
        {
          source: "/",
          destination: "/",
          basePath: false,     // This somehow isnt recognized as an option.
        },
      ];
    }
    return [];
  },...
}
```

**Comment by IikkaWinter** [comment]
I can confirm that basePath property has no effect and makes it impossible to add a route to the root page '/'.
This issue together with assetPrefixing not working for getStaticProps (https://github.com/vercel/next.js/issues/16090) - and therefore requiring basePath - is now putting us in a position without a solution.
